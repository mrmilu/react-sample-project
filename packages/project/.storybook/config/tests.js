import { addDecorator } from '@storybook/react';
import { withTests } from '@storybook/addon-jest';

import results from '../../src/stories/.jest-test-results.json';

addDecorator(withTests({ results }));
