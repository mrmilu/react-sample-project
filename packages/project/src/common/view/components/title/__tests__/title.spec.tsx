import React from 'react';
import { customRender } from '../../../../../../tests/test-utils';
import Title from '../title.component';

describe('Title', () => {
  it('Render', () => {
    const rendered = customRender(<Title>The title</Title>);
    const item = rendered.getByText('The title');
    expect(item).not.toBeNil();
    expect(rendered.container).toMatchSnapshot();
  });
  it('Should not contains bold className', () => {
    const rendered = customRender(<Title>The title</Title>);
    const result = rendered.container.querySelector('.bold');
    expect(result).toBeNil();
  });
  it('Should contains bold className', () => {
    const rendered = customRender(<Title bold>The title</Title>);
    const result = rendered.container.querySelector('.bold');
    expect(result).not.toBeNil().toBeObject();
    expect(rendered.container).toMatchSnapshot();
  });
});
