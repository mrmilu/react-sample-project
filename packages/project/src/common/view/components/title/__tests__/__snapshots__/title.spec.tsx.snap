// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Render 1`] = `
<div>
  <h2
    class=""
  >
    The title
  </h2>
</div>
`;

exports[`Title Should contains bold className 1`] = `
<div>
  <h2
    class="bold"
  >
    The title
  </h2>
</div>
`;
